astropy/io/fits/scripts/fitsdiff.py

```diff
@@ -80,7 +80,8 @@ def __call__(self, parser, namespace, values, option_string=None):
                 log.warning(f"{self.dest} argument {value} does not exist")
                 return
             try:
-                values = [v.strip() for v in open(value).readlines()]
+                with open(value) as fd:
+                    values = fd.read().splitlines()
```

Comment:
I should note that I refactored this manually for consistency with other similarly affected snippets, but this particular one isn't covered by tests, and the new proposed code isn't strictly equivalent in that it reads the whole file eagerly. Happy to revert if this feels risky to package maintainers.

Reply:
Well, technically, strip a line before it's stored would still be more memory effective than storing everything, then strip. The first implementation barely requires more memory than the final result, while this refactored version might consume up to a bit over twice as much, but in practice it really shouldn't matter.

Reply:
`readlines()` returns a `list`, so that's reading eagerly too.

Reply:
This is not even reading a data file, just a list of header keys or comments to be ignored, so should not be any problem to read the whole thing at once.

Reply:
But comparing with how the values from the command line are set in L92, the previous code would be robust against files with blanks (e.g. for alignment of the entries), while this version is not.

Reply:
This refactor was moved to #16656. Please consider closing this thread.